### ISSUE TITLE
Use Case: medical image segmentation

### ISSUE THREAD

**Issue by lorenzoh** [FluxML/ML-Coordination-Tracker#15]
This use case covers [this MonAI tutorial](https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/brats_segmentation_3d.ipynb) for 3D multi-class semantic segmentation.

Below is a list of references to parts that can be done and functionality that is missing to recreate the tutorial in Julia. 

## Dataset/data container

Available from http://medicaldecathlon.com/

ToDos:
- create a data container (see https://lorenzoh.github.io/DataLoaders.jl/dev/docs/datacontainers.html)
- one observation is one 3D image and multiple 3D masks
- select orientation 3d image is read in

## Data preprocessing/transformation/augmentation

- crop/resize/scale
- normalization
- random jitter

See [DataAugmentation.jl](https://github.com/lorenzoh/DataAugmentation.jl)

ToDos:
- item implementation for 3d iamges and masks
- affine transformations support 3D images and 3D masks
- random jitter transform

## Data loading/data iterator

Use DataLoaders.jl

For wrapping the whole data pipeline, consider [DLPipelines.jl](https://lorenzoh.github.io/DLPipelines.jl/dev/README.html)

## Model

see https://github.com/DhairyaLGandhi/UNet.jl, should probably work fine. Maybe add some tweaks from fastai's implementation.

## Training

- dice loss function
- optimizer: use `Flux.ADAM`
- training loop: use FluxTraining.jl
- for logs and visualizations, use `Logger` with `TensorBoardBackend`
- use `Checkpointer` to save the model

ToDos

- implement dice in FluxTraining.jl

**Comment by darsnack** [comment]
I am closing this after moving it to [a discussion topic](https://github.com/FluxML/ML-Coordination-Tracker/discussions/20). The idea is to reduce the issues section to actionable items for contributors. Discussions and user stories can go in the Discussions tab.
